src/mult.jl

```diff
@@ -29,6 +29,17 @@ include("cmult.jl")
 include("ncmult.jl")
 
 MP.multconstant(α, x::Monomial)   = MP.term(α, MA.mutable_copy(x))
+
+function MP.multconstant(α, p :: Polynomial )
```

Comment:
I would define a function `zero_with_variables` and do
```julia
function _multconstant(α::T, f, p::AbstractPolynomial{S}) where {S, T}
    if iszero(α)
        zero_with_variables(polynomialtype(p, MA.promote_operation(*, T, S)), variables(p))
    else
        mapcoefficientsnz(f, p)
    end
end
```

Reply:
As suggested I have defined `zero_with_variables` as 
```
function zero_with_variables( ::Type{Polynomial{C,T}}, vars :: Vector{PolyVar{C}} ) where{C, T}
    Polynomial( T[], emptymonovec(vars) )
end
```
Overwriting `_multconstant` via 
```
function MP._multconstant(α::T, f, p::AbstractPolynomial{S}) where {S, T}
    if iszero(α)
        zero_with_variables(polynomialtype(p, MA.promote_operation(*, T, S)), variables(p))
    else
        mapcoefficientsnz(f, p)
    end
end
```
works but breaks something ( `** incremental compilation may be fatally broken for this module **` )

Reply:
It means you have defined twice the same method, remove the other one you have defined.

Reply:
I did change the function signature to apply to DynamicPolynomials  only, i.e., `MP._multconstant(α::T, f, p::Polynomial{C,S} ) where {T, C, S}` 

That does the trick